src/routes/scope-selection/ScopeSelection.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useMemo } from 'react'
+import styled, { useTheme } from 'styled-components/native'
+
+import DisciplineListItem from '../../components/DisciplineListItem'
+import SearchBar from '../../components/SearchBar'
+import ServerResponseHandler from '../../components/ServerResponseHandler'
+import { ContentTextBold, ContentTextLight } from '../../components/text/Content'
+import { Discipline } from '../../constants/endpoints'
+import { formatDiscipline } from '../../hooks/helpers'
+import { useLoadAllDisciplines } from '../../hooks/useLoadAllDisciplines'
+import { useLoadDisciplines } from '../../hooks/useLoadDisciplines'
+import { pushSelectedProfession } from '../../services/AsyncStorage'
+import { getLabels, searchProfessions, splitTextBySearchString } from '../../services/helpers'
+
+const SearchContainer = styled.View`
+  margin: ${props => props.theme.spacings.sm};
+`
+
+const ScopeContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
+  padding: ${props => props.theme.spacings.xs} 0;
+  background-color: ${props => props.theme.colors.background};
+`
+
+const StyledPressable = styled.Pressable`
+  padding: ${props => props.theme.spacings.sm};
+  border-bottom: 1px solid ${props => props.theme.colors.disabled};
+`
+
+const DisciplineContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
+`
+
+const HighlightContainer = styled.Text<{ disabled: boolean }>`
+  flex-direction: row;
+  color: ${props => (props.disabled ? props.theme.colors.disabled : props.theme.colors.black)};
+`
+
+const highlightText = (textArray: [string] | [string, string, string], disabled: boolean): JSX.Element => (
+  <HighlightContainer disabled={disabled}>
+    <ContentTextLight>{textArray[0]}</ContentTextLight>
+    <ContentTextBold>{textArray[1]}</ContentTextBold>
+    <ContentTextLight>{textArray[2]}</ContentTextLight>
+  </HighlightContainer>
+)
+
+type ScopeSelectionProps = {
+  queryTerm: string
+  setQueryTerm: (newString: string) => void
+  showSearchResults: boolean
+  setShowSearchResults: (shouldShow: boolean) => void
+  navigateToDiscipline: (item: Discipline) => void
+  navigateToManageSelection: () => void
+  selectedProfessions: number[] | null
+}
+
+const ScopeSelection = ({
+  queryTerm,
+  setQueryTerm,
+  showSearchResults,
+  setShowSearchResults,
```

Comment:
:wrench: AFAICS `showSearchResults` is always depending on the length of the `queryTerm`, i.e. if the query term is not empty it is true and otherwise false. Therefore this is redundant and could be removed.

Reply:
Good catch, thanks!